src/queueing_honey_badger.rs

```diff
@@ -154,9 +155,7 @@ where
             self.queue.remove_all(batch.iter());
             self.output.push_back(batch);
         }
-        if !self.dyn_hb.has_input() {
-            self.propose()?.merge_into(&mut fault_log);
-        }
+        self.propose()?.merge_into(&mut fault_log);
```

Comment:
I think we are getting rid of `merge_into()` in exchange for `fault_log.extend(self.propose()?);`

Reply:
Yes, there are still plenty of `merge_into` calls. I'd prefer to replace them in a separate PR.
(I didn't change this line; only removed indentation, that's why I missed it).

Reply:
Alright, sounds good.